homework_3/events/index.js

```diff
@@ -0,0 +1,97 @@
+const stream = require('stream');
+const fs = require('fs');
+const path = require('path');
+const csv = require('csv-parser');
+
+// --- --- --- --- --- --- --- Handlers --- --- --- --- --- --- --- ---
+
+getAll = async(req, res) => readEvents().then(events => {
+    const streamEvents = stream.Readable.from(JSON.stringify(events));
+    streamEvents.pipe(res);
+});
+
+getEvents = async(req, res) => {
+    const location = req.query.location;
+    readEvents().then(events => {
+        if (location)
+            events = events.filter(event => event.location === location);
+        res.json(events);
+    });
+}
+
+getEvent = async(req, res) => {
+    const id = Number(req.params.id);
+    if (Number.isInteger(id)) {
+        readEvents().then(events => res.json(events[id]));
```

Comment:
`events[id]` - it looks like we are grepping event by its index in array/file, not by `id` field